File: frontend/src/api/hfs.ts
import http from './http'

/** POST /api/hfs/upload/create */
export function createUpload(data: {
  fileName: string
  fileSize: number
  chunkSize: number
  totalChunks: number
}) {
  return http.post<{ uploadId: string; tempDir: string }>('/hfs/upload/create', data)
}

/** PUT /api/hfs/upload/chunk */
export function uploadChunk(uploadId: string, chunkIndex: number, blob: Blob, chunkMd5?: string) {
  const form = new FormData()
  form.append('file', blob)

  const params = new URLSearchParams()
  params.set('uploadId', uploadId)
  params.set('chunkIndex', String(chunkIndex))
  if (chunkMd5) params.set('chunkMd5', chunkMd5)

  return http.put<{ status: string }>(`/hfs/upload/chunk?${params.toString()}`, form, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

/** POST /api/hfs/upload/merge */
export function mergeUpload(uploadId: string) {
  return http.post<{
    uploadId: string
    filePath: string
    fileName: string
    fileSize: number
  }>('/hfs/upload/merge', { uploadId })
}

/** POST /api/hfs/assets — 将合并后的临时文件转为静态资源 */
export function commitAssets(uploadId: string) {
  return http.post<{ path: string }>('/hfs/assets', { uploadId })
}

File: backend/vo/hfs.go
package vo

// ChunkUploadCreateReq 创建上传任务请求
type ChunkUploadCreateReq struct {
	FileName    string `json:"fileName" validate:"required"`
	FileSize    int64  `json:"fileSize" validate:"required"`
	ChunkSize   int    `json:"chunkSize" validate:"required"`
	TotalChunks int    `json:"totalChunks" validate:"required"`
}

// ChunkUploadCreateRsp 创建上传任务响应
type ChunkUploadCreateRsp struct {
	UploadId string `json:"uploadId"`
	TempDir  string `json:"tempDir"`
}

// ChunkUploadReq 上传分片请求
type ChunkUploadReq struct {
	UploadId   string `form:"uploadId" validate:"required"`
	ChunkIndex int    `form:"chunkIndex"`
	ChunkMd5   string `form:"chunkMd5"`
}

// ChunkMergeReq 合并上传请求
type ChunkMergeReq struct {
	UploadId string `json:"uploadId" validate:"required"`
}

// ChunkMergeRsp 合并上传响应
type ChunkMergeRsp struct {
	UploadId string `json:"uploadId"`
	FilePath string `json:"filePath"`
	FileName string `json:"fileName"`
	FileSize int64  `json:"fileSize"`
}

// AssetsReq 提交静态资源请求
type AssetsReq struct {
	UploadId string `json:"uploadId" validate:"required"`
}

// AssetsRsp 提交静态资源响应
type AssetsRsp struct {
	Path string `json:"path"`
}

// TempAccessReq 申请临时访问凭证请求
type TempAccessReq struct {
	Path string `json:"path" validate:"required"` // 用户空间相对路径
	Type string `json:"type" validate:"required"` // "file" 或 "dir"
}

// TempAccessRsp 申请临时访问凭证响应
type TempAccessRsp struct {
	Ak        string `json:"ak"`
	ExpiresAt int64  `json:"expiresAt"` // 过期时间（unix 秒）
}
